src/rgw/rgw_op.cc

```diff
@@ -1615,6 +1615,7 @@ static int iterate_user_manifest_parts(CephContext * const cct,
   rgw::sal::RGWBucket::ListParams params;
   params.prefix = obj_prefix;
   params.delim = delim;
+  params.shard_id = -1;
```

Comment:
it feels like this should be the default behavior for `ListParams`, doesn't it? can we update the value `int shard_id{0};` where it's declared in rgw_sal.h?

cc @dang

Reply:
oh, you guys are already on it! thanks